cmake/AwsCheckHeaders.cmake

```diff
@@ -15,57 +15,84 @@ endif()
 
 # Call as: aws_check_headers(${target} HEADERS TO CHECK LIST)
 function(aws_check_headers target)
+    # parse function arguments
+    set(options DISABLE_C_HEADER_CHECKER)
+    cmake_parse_arguments(ARG "${options}" "" "" ${ARGN})
+
     if (PERFORM_HEADER_CHECK)
+        aws_check_headers_internal(${target} 11 ${ARG_DISABLE_C_CHECKER} ${ARG_UNPARSED_ARGUMENTS})
+        aws_check_headers_internal(${target} 14 ${ARG_DISABLE_C_CHECKER} ${ARG_UNPARSED_ARGUMENTS})
+        aws_check_headers_internal(${target} 17 ${ARG_DISABLE_C_CHECKER} ${ARG_UNPARSED_ARGUMENTS})
+        aws_check_headers_internal(${target} 20 ${ARG_DISABLE_C_CHECKER} ${ARG_UNPARSED_ARGUMENTS})
+        aws_check_headers_internal(${target} 23 ${ARG_DISABLE_C_CHECKER} ${ARG_UNPARSED_ARGUMENTS})
+    endif() # PERFORM_HEADER_CHECK
+endfunction()
 
-        set(HEADER_CHECKER_ROOT "${CMAKE_CURRENT_BINARY_DIR}/header-checker")
+function(aws_check_headers_internal target std disable_c_header_checker)
+    # Check that compiler supports this std
+    list (FIND CMAKE_CXX_COMPILE_FEATURES "cxx_std_${std}" feature_idx)
+    if (${feature_idx} LESS 0)
+        return()
+    endif()
 
-        # Write stub main file
+    set(HEADER_CHECKER_ROOT "${CMAKE_CURRENT_BINARY_DIR}/header-checker-${std}")
+
+    # Write stub main file
+    if(disable_c_header_checker)
+        set(HEADER_CHECKER_MAIN "${HEADER_CHECKER_ROOT}/headerchecker_main.cpp")
+        set(HEADER_CHECKER_LIB ${target}-header-check-cxx-${std})
+    else()
         set(HEADER_CHECKER_MAIN "${HEADER_CHECKER_ROOT}/headerchecker_main.c")
-        file(WRITE ${HEADER_CHECKER_MAIN} "
-            int main(int argc, char **argv) {
-                (void)argc;
-                (void)argv;
+        set(HEADER_CHECKER_LIB ${target}-header-check-${std})
```

Comment:
Did this really have to change? I'm pretty sure it's fine for a C++ application to have a main.c

Reply:
Thanks, fixed.